source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -445,30 +481,76 @@ HttpConnectionManagerConfig::HttpConnectionManagerConfig(
 void HttpConnectionManagerConfig::processFilter(
     const envoy::extensions::filters::network::http_connection_manager::v3::HttpFilter&
         proto_config,
-    int i, absl::string_view prefix, std::list<Http::FilterFactoryCb>& filter_factories,
+    int i, absl::string_view prefix, FilterFactoriesList& filter_factories,
     const char* filter_chain_type, bool last_filter_in_current_config) {
   ENVOY_LOG(debug, "    {} filter #{}", prefix, i);
   ENVOY_LOG(debug, "      name: {}", proto_config.name());
-  ENVOY_LOG(debug, "    config: {}",
-            MessageUtil::getJsonStringFromMessage(
-                proto_config.has_typed_config()
-                    ? static_cast<const Protobuf::Message&>(proto_config.typed_config())
-                    : static_cast<const Protobuf::Message&>(
-                          proto_config.hidden_envoy_deprecated_config()),
-                true));
-
-  // Now see if there is a factory that will accept the config.
-  auto& factory =
-      Config::Utility::getAndCheckFactory<Server::Configuration::NamedHttpFilterConfigFactory>(
-          proto_config);
-  ProtobufTypes::MessagePtr message = Config::Utility::translateToFactoryConfig(
-      proto_config, context_.messageValidationVisitor(), factory);
-  Http::FilterFactoryCb callback =
-      factory.createFilterFactoryFromProto(*message, stats_prefix_, context_);
-  bool is_terminal = factory.isTerminalFilter();
-  Config::Utility::validateTerminalFilters(proto_config.name(), factory.name(), filter_chain_type,
-                                           is_terminal, last_filter_in_current_config);
-  filter_factories.push_back(callback);
+  switch (proto_config.config_type_case()) {
+  case envoy::extensions::filters::network::http_connection_manager::v3::HttpFilter::
```

Comment:
This code is now pretty long and complicated with lots of if and nesting. Could you restructure for readability? Thanks.

Reply:
Done.